src/systems/filecoin_vm/interpreter/_index.md

```diff
@@ -50,6 +50,36 @@ The sequence of executions for a tipset is thus summarised:
 - [... subsequent blocks ...]
 - cron tick 
 
+# Message validity and failure
+Every syntactically valid message can be processed and produce a receipt. However, execution may
+or may not succeed, depending on the state to which the message is applied. If the execution
+of a message fails, the corresponding receipt will carry a non-zero exit code, and the only
+state changes that will persist will be:
+
+- incrementing of the sending actor's `CallSeqNum` (if the actor exists), and
+- payment of gas fees, either from the sender to the owner of the miner of the block including the message, or burnt by the miner.
+ 
+The sender pays the gas fee to the miner if and only if sending actor's `CallSeqNum` is incremented.
+Otherwise, the miner pays (burns) the gas fee as penalty for including the failing message in a block.
+ 
+A message execution will fail if:
+
+- the `From` actor does not exist in the state (miner penalized),
+- the `From` actor is not an account actor (miner penalized),
+- the `CallSeqNum` of the message does not match the `CallSeqNum` of the `From` actor (miner penalized),
+- the `To` actor does not exist in state and the `To` address is not an pubkey-style address (miner penalized),
+- the `To` actor does not exist in state and the message has a non-zero `Method` (miner penalized),
+- the `To` actor does not have a method corresponding to `MethodNum`,
+- deserialized `Params` is not an array of length matching the arity of the `To` actor's `MethodNum` method,
+- deserialized `Params` are not valid for the types specified by the `To` actor's `MethodNum` method,
+- the `From` actor does not have sufficient balance to cover the sum of the message `Value` plus the
+maximum gas cost, `GasLimit * GasPrice` (miner penalized),
+- the invoked method consumes more gas than the `GasLimit` allows, or
+- the invoked method exits with a non-zero code (via `Runtime.Abort()`).
+
+Note that if the `To` actor does not exist in state and the address is a valid `H(pubkey)` address, 
+it will be created as an account actor (only if the message has a MethodNum of zero).
+
```

Comment:
also worth giving a heuristic at the top of when the miner is penalized vs not. Looks a bit random without that explanation up front.